cmd/delete_test.go

```diff
@@ -8,6 +8,61 @@ import (
 	"knative.dev/func/mock"
 )
 
+// TestDelete_Namespace ensures that the namespace provided to the client
+// for use when deleting a function is set
+// 1. The flag /env variable if provided
+// 2. The namespace of the function at path if provided
+// 3. The user's current active namespace
+func TestDelete_Namespace(t *testing.T) {
+	root := fromTempDirectory(t)
+
+	// Ensre that the default is "default" when no context can be identified
+	cmd := NewDeleteCmd(func(cc ClientConfig, options ...fn.Option) (*fn.Client, func()) {
+		if cc.Namespace != "default" {
```

Comment:
Will this work if developer is logged in non-default namespace (or in future has some override in config)?

Reply:
```
=== RUN   TestDelete_Namespace
    delete_test.go:22: expected 'default', got 'mvasek-dev'
    delete_test.go:44: expected 'deployed', got 'mvasek-dev'
--- FAIL: TestDelete_Namespace (0.01s)
```

Reply:
and even with clean config:
```
    delete_test.go:44: expected 'deployed', got 'default'
 ```

Reply:
How are test passing in CI?

Reply:
You need to set args to empty `cmd.SetArgs([]string{})` on line 48.

Reply:
So it works from IDE.

Reply:
Also you could add:
```go
t.Setenv("KUBECONFIG", "/tmp/non-existent")
```
at beginning of the test to avoid unwanted interaction with developers config.

Reply:
Ignoring current user's KUBECONFIG and clearing args from IDEs are now being addressed in separate PRs